src/Lachain.Core/Blockchain/Validators/ValidatorManager.cs

```diff
@@ -13,6 +13,7 @@ public class ValidatorManager : IValidatorManager
     {
         private readonly ISnapshotIndexRepository _snapshotIndexRepository;
         private static readonly ILogger<ValidatorManager> Logger = LoggerFactory.GetLoggerForClass<ValidatorManager>();
+        private static Dictionary<long, IReadOnlyCollection<ECDSAPublicKey>> _pubkeyCache = new Dictionary<long, IReadOnlyCollection<ECDSAPublicKey>>();
```

Comment:
Is it valid to make it static in this situation?

Reply:
It's technically valid provided this class is used as singleton (it currently is). However cleaner solution IMO would be
`private readonly Dictionary<long, IReadOnlyCollection<ECDSAPublicKey>> _pubkeyCache`
This way one can create multiple instances of class (e.g. may be useful in tests) which won't interfere

Reply:
Fixed